jrt/src/com/yahoo/jrt/SslSocket.java

```diff
@@ -179,6 +181,8 @@ private int sslEngineUnwrap(ByteBuffer dst) throws IOException {
         switch (status) {
             case OK:
                 return result.bytesProduced();
+            case BUFFER_OVERFLOW:
+                throw new SSLException("Cannot unwrap - remaining capacity too small: " + dst);
```

Comment:
You cannot do this as long as you have a loop around the call to unwrap. A channel read could read 2 frames (since there is extra room in the input buffer) and the output buffer could be too small even if it adheres to the buffer size requirements.

Reply:
Good catch - it should return 0 instead.

Reply:
I have updated unwrap logic to use separate code path for handling overflow. Overflow will no longer throw exception.